plugin/pkg/parse/host.go

```diff
@@ -33,6 +33,15 @@ func HostPortOrFile(s ...string) ([]string, error) {
 	var servers []string
 	for _, h := range s {
 		trans, host := Transport(h)
+		if len(host) == 0 {
+			return servers, fmt.Errorf("invalid address: %q", h)
+		}
+
+		switch trans {
```

Comment:
Use the `if` statement for a single case.

Reply:
see plugin/pkg/parse/host_test.go:66

Reply:
I meant that comparing only the `transport.UNIX` case, it would be better to use an `if` statement instead of `swtich`.

```
if trans == transport.UNIX {
	servers = append(servers, trans+"://"+host)
        continue
}
```

Reply:
I am accustomed to use `switch`, but i accept your propose and will submit it later.

Reply:
PTAL

Reply:
It’s really strange.I already had this modified and submitted it. I even double-checked it.

Reply:
@inigohu